lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -355,6 +356,279 @@ def test_different_array_attrs_incompatible(self):
         self.assertFalse(self.test_coord.is_compatible(self.other_coord))
 
 
+class Test_contiguous_bounds(tests.IrisTest):
+    def test_no_bounds_warning(self):
+        coord = iris.coords.DimCoord([0, 1, 2], standard_name='latitude')
+        msg = "Coordinate 'latitude' is not bounded, guessing contiguous " \
+              "bounds."
+        with warnings.catch_warnings():
+            # Cause all warnings to raise Exceptions
+            warnings.simplefilter("error")
+            with self.assertRaisesRegexp(Warning, msg):
+                coord.contiguous_bounds()
+
+    def test_1d_coord(self):
+        coord = iris.coords.DimCoord([2, 4, 6], standard_name='latitude',
+                                     bounds=[[1, 3], [3, 5], [5, 7]])
+        expected = np.array([1, 3, 5, 7])
+        result = coord.contiguous_bounds()
+        self.assertArrayEqual(result, expected)
+
+    def test_1d_coord_discontiguous(self):
+        coord = iris.coords.DimCoord([2, 4, 6], standard_name='latitude',
+                                     bounds=[[1, 3], [4, 5], [5, 7]])
+        expected = np.array([1, 4, 5, 7])
+        result = coord.contiguous_bounds()
+        self.assertArrayEqual(result, expected)
```

Comment:
I didn't include tests for 2d here is because `is_contiguous` is effectively a wrapper for calling `_discontiguity_in_bounds` and has things were already being tested in `Test__discontiguity_in_bounds` it seemed repetitive to also have them here.

I added the 1d cases to show that the behaviour is as before (before 1d was handled in `_discontiguity_in_bounds`.

I will add a couple of simple 2d examples and hopefully that will be enough testing.